modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -192,8 +193,8 @@ public DataStreamLifecycleService(
         );
         this.swapSourceWithDownsampleIndexQueue = clusterService.createTaskQueue(
             "data-stream-lifecycle-swap-source-with-downsample",
-            Priority.NORMAL,
-            new ReplaceBackingWithDownsampleIndexExecutor(client)
+            Priority.URGENT, // urgent priority as this deletes indices
```

Comment:
What makes this urgent? I'm not arguing against it, I've just never seen any guidelines so I'm curious.

Reply:
Anecdotally anything operating on the index state is urgent (i.e. open/close/delete). I don't think we have better guidelines, but I've been following this.